test/goalc/test_with_game.cpp

```diff
@@ -345,6 +342,219 @@ TEST_F(WithGameTests, StaticBoxedArray) {
+        return z;
+      case emitter::Register::VF_ELEMENT::W:
+        return w;
+      default:
+        return defValue;
+    }
+  }
+
+  std::string toGOALFormat() {
+    std::string answer = fmt::format("({:.4f}, {:.4f}, {:.4f}, {:.4f})", x, y, z, w);
+    // {fmt} formats negative 0 as "-0.000", just going to flip any negative zeros to positives as I
+    // don't think is an OpenGOAL issue
+    return std::regex_replace(answer, std::regex("-0.0000"), "0.0000");
+  }
+};
+
+struct VectorFloatTestCase {
+  VectorFloatRegister input1 = {1.5, -1.5, 0.0, 100.5};
+  VectorFloatRegister input2 = {-5.5, -0.0, 10.0, 7.5};
+  VectorFloatRegister dest = {11, 22, 33, 44};
+
+  int destinationMask = -1;
+  emitter::Register::VF_ELEMENT bc = emitter::Register::VF_ELEMENT::NONE;
+  std::function<float(float, float)> operation;
+
+  VectorFloatRegister getExpectedResult() {
+    VectorFloatRegister expectedResult;
+    expectedResult.x = destinationMask & 0b0001
+                           ? operation(input1.x, input2.getBroadcastElement(bc, input2.x))
+                           : dest.x;
+    expectedResult.y = destinationMask & 0b0010
+                           ? operation(input1.y, input2.getBroadcastElement(bc, input2.y))
+                           : dest.y;
+    expectedResult.z = destinationMask & 0b0100
+                           ? operation(input1.z, input2.getBroadcastElement(bc, input2.z))
+                           : dest.z;
+    expectedResult.w = destinationMask & 0b1000
+                           ? operation(input1.w, input2.getBroadcastElement(bc, input2.w))
+                           : dest.w;
+    return expectedResult;
+  }
+
+  std::string getOperationBroadcast() {
+    switch (bc) {
+      case emitter::Register::VF_ELEMENT::X:
+        return "x";
+      case emitter::Register::VF_ELEMENT::Y:
+        return "y";
+      case emitter::Register::VF_ELEMENT::Z:
+        return "z";
+      case emitter::Register::VF_ELEMENT::W:
+        return "w";
+      default:
+        return "";
+    }
+  }
+
+  void setJson(nlohmann::json& data, std::string func, bool twoOperands = true) {
+    input1.setJson(data, "v1");
+    data["twoOperands"] = twoOperands;
+    if (twoOperands) {
+      input2.setJson(data, "v2");
+    }
+    dest.setJson(data, "dest");
+    data["operation"] = fmt::format(func);
+    if (destinationMask == -1) {
+      data["destinationMask"] = false;
+    } else {
+      data["destinationMask"] = fmt::format("{:b}", destinationMask);
+    }
+  }
+};
+
+std::vector<VectorFloatTestCase> vectorMathTestCaseGen() {
+  std::string test = fmt::format("{:.4f}", -0.0);
+
+  std::vector<VectorFloatTestCase> cases = {};
+  for (int i = 0; i <= 15; i++) {
+    VectorFloatTestCase testCase = VectorFloatTestCase();
+    testCase.destinationMask = i;
+    cases.push_back(testCase);
+    // Re-add each case with each broadcast varient
+    for (int j = 0; j < 4; j++) {
+      VectorFloatTestCase testCase = VectorFloatTestCase();
```

Comment:
compiler warning on `testCase` here having the same name as `testCase` on line 445